src/pint/models/binary_piecewise.py

```diff
@@ -0,0 +1,325 @@
+                    description_template=lambda x: param,
+                    parameter_type="float",
+                    frozen=False,
+                )
+            )
+
+        # self.setup()
+
+    def setup(self):
+        # print(self.PB)
+        super(BinaryBTPiecewise, self).setup()
+        # print(self.PB)
+        for bpar in self.params:
+            self.register_deriv_funcs(self.d_binary_delay_d_xxxx, bpar)
+        # Setup the model isinstance
+        self.binary_instance = self.binary_model_class()
+        # Setup the FBX orbits if FB is set.
+        # TODO this should use a smarter way to set up orbit.
+        T0X_mapping = self.get_prefix_mapping_component("T0X_")
+        # print(T0X_mapping)
+        T0Xs = {}
+        A1X_mapping = self.get_prefix_mapping_component("A1X_")
+        A1Xs = {}
+        XR1_mapping = self.get_prefix_mapping_component("PLB_")
+        XR1s = {}
+        XR2_mapping = self.get_prefix_mapping_component("PUB_")
+        XR2s = {}
+        # print(f'XR2:{XR2_mapping.values()}')
+        for t0n in T0X_mapping.values():
+            # print("hello from T0 mapping")
+            T0Xs[t0n] = getattr(self, t0n).quantity
+        # if None in T0Xs.values():
+        # print("Group with non-defined T0X, applying default T0 to group")
+        # TODO set default T0 value
+        if None not in T0Xs.values():
+            for t0_name, t0_value in T0Xs.items():
+                self.binary_instance.add_binary_params(t0_name, t0_value)
+
+        for a1n in A1X_mapping.values():
+            # print("hello from A1 mapping")
+            A1Xs[a1n] = getattr(self, a1n).quantity
+
+        # if None in A1Xs.values():
+        # print("Group with non-defined A1X, applying default A1 to group")
+        # TODO set default A1 value
+
+        if None not in A1Xs.values():
+            # print(len(A1Xs.items()))
+            for a1_name, a1_value in A1Xs.items():
+                self.binary_instance.add_binary_params(a1_name, a1_value)
+        #
+        for XR1n in XR1_mapping.values():
+            # lower bound mapping
+            # print("hello from Lower bound mapping mapping")
+            XR1s[XR1n] = getattr(self, XR1n).quantity
+
+        if None not in XR1s.values():
+            for xr1_name, xr1_value in XR1s.items():
+                self.binary_instance.add_binary_params(xr1_name, xr1_value)
+
+        for XR2n in XR2_mapping.values():
+            XR2s[XR2n] = getattr(self, XR2n).quantity
+        # print(XR1s)
+        # if None in XR2s.values():
+        # print("Group with non-defined XR2, applying default A1 to group")
+        # TODO set default A1 value
+
+        if None not in XR2s.values():
+            # print(len(A1Xs.items()))
+            for xr2_name, xr2_value in XR2s.items():
+                self.binary_instance.add_binary_params(xr2_name, xr2_value)
+        self.update_binary_object()
+
+    def validate(self):
```

Comment:
Are you sure there aren't any ways the new parameters could be set that would make the model unusable? `validate()` should check for invalid settings of the piecewise parameters. (Overlapping T0X intervals? T0X intervals with end before beginning? Any weirdness that won't work can be detected here and then you won't need to worry about it later.)